### PR TITLE
expression: add json_merge_patch

### PR DESCRIPTION
**Pull request by yaochengji**
Thanks for reviewing in advance.

### Release Note
- added `JSONMergePatch`

### CODE REVIEW

**Comment by yaochengji** [comment]
@lzmhhh123 could you help review?

**Comment by zz-jason** [comment]
@XuHuaiyu PTAL

File: types/json/binary_functions.go [hunk 1]
```diff
@@ -763,6 +763,59 @@ func MergeBinary(bjs []BinaryJSON) BinaryJSON {
...
+	for i := 1; i < len(bjs); i++ {
+		result = mergePatchBinaryObject(&result, &(bjs[i]))
+	}
+	return result
+}
+
+func mergePatchBinaryObject(oldObj *BinaryJSON, newObj *BinaryJSON) BinaryJSON {
+	if newObj.TypeCode != TypeCodeObject {
+		return *newObj
+	}
```
Comment (lzmhhh123) [inline types/json/binary_functions.go:780]: I feel confused about this logic. The `MergeJson` function in golang would merge the multiple arrays into one array.

**Comment by tisonkun** [comment]
Closed as stale.

@yaochengji you can create an issue first and discuss on the motivation and design instead of rushing a PR which is less likely to be reviewed.

**Pull request closed without merging**